javascript/ql/lib/ext/underscore.string.model.yml

```
@@ -4,3 +4,4 @@ extensions:
       extensible: summaryModel
     data:
       - ["'underscore.string'", "Member[slugify,capitalize,decapitalize,clean,cleanDiacritics,swapCase,escapeHTML,unescapeHTML,wrap,dedent,reverse,pred,succ,titleize,camelize,classify,underscored,dasherize,humanize,trim,ltrim,rtrim,truncate,sprintf,strRight,strRightBack,strLeft,strLeftBack,stripTags,unquote,map]", "Argument[0]", "ReturnValue", "taint"]
+      - ["'underscore.string'", "Member[chop,chars,words,lines]", "Argument[0]", "ReturnValue", "taint"]
```

```suggestion
      - ["'underscore.string'", "Member[chop,chars,words,lines]", "Argument[0]", "ReturnValue.ArrayElement", "taint"]
```
This is a bit more precise.

If the whole return value is tainted, the analysis thinks all its properties are tainted too, including `.length`. You could try adding the test:
```js
sink(s.chop(source("s1"), 3).length);
```

---

It looks like there is still something off with `ArrayElement` [f4ca2dc](https://github.com/github/codeql/pull/19049/commits/f4ca2dc1f30a445a9cc136b62b61ab16b35f309e)?

---

Some of the tests have a typo, they should access the `[0]` on the returned array, not on the source string:
```js
sink(s.chars(source("s2")[0])) -> sink(s.chars(source("s2"))[0])
```

I know why the last test is failing but it's complicated. Would you mind leaving it as `MISSING:` for now?